clientconn_parsed_target_test.go

```diff
@@ -132,15 +131,34 @@ func (s) TestParsedTarget_WithCustomDialer(t *testing.T) {
 			wantParsed:        resolver.Target{Scheme: "unix", Authority: "", Endpoint: "/a/b/c"},
 			wantDialerAddress: "unix:///a/b/c",
 		},
+		{
+			target:            "dns:///127.0.0.1:50051",
+			wantParsed:        resolver.Target{Scheme: "dns", Authority: "", Endpoint: "127.0.0.1:50051"},
+			wantDialerAddress: "127.0.0.1:50051",
+		},
+		{
+			target:            ":///127.0.0.1:50051",
+			wantParsed:        resolver.Target{Scheme: defScheme, Authority: "", Endpoint: ":///127.0.0.1:50051"},
+			wantDialerAddress: ":///127.0.0.1:50051",
+		},
+		{
+			target:            "dns://authority/127.0.0.1:50051",
+			wantParsed:        resolver.Target{Scheme: "dns", Authority: "authority", Endpoint: "127.0.0.1:50051"},
+			wantDialerAddress: "127.0.0.1:50051",
+		},
+		{
+			target:            "://authority/127.0.0.1:50051",
+			wantParsed:        resolver.Target{Scheme: defScheme, Authority: "", Endpoint: "://authority/127.0.0.1:50051"},
+			wantDialerAddress: "://authority/127.0.0.1:50051",
+		},
 	}
 
 	for _, test := range tests {
 		t.Run(test.target, func(t *testing.T) {
+			addrCh := make(chan string, 1)
 			dialer := func(ctx context.Context, address string) (net.Conn, error) {
-				if address != test.wantDialerAddress {
-					return nil, fmt.Errorf("address in custom dialer is %q, want %q", address, test.wantDialerAddress)
-				}
-				return (&net.Dialer{}).DialContext(ctx, "", address)
+				addrCh <- address
```

Comment:
I'm slightly concerned this could lead to a goroutine hang if this gets called three times before we close the ClientConn.  You might be able to stimulate this by putting in a long enough sleep into the test after the select where we read from the channel.  Blocking on the context inside a select here should remove that possibility.

Maybe this is practically impossible given the timings (a long enough backoff), but it still seems worthwhile.

Reply:
As per our offline discussion, the `subConn` in this case moves to `Idle` and therefore there are no retries here. So, we should be good. Thanks.